fix(max): stop autoadding context

kappa90 opened the pull request:
## Problem
`maxContextLogic` had a bug for which scene context would still be added even if manually removed.

Since we already had a [discussion](https://posthog.slack.com/archives/C06NZEZ7V3Q/p1753339257658279) about potentially removing this functionality, I decided to just remove it. You can still add scene context, manually.

## Changes
- Removed logic to auto add scene UI context

## How did you test this code?
Locally

Commit:
fix(max): stop autoadding context

skoob13 reviewed (changes requested):
The context doesn't automatically attach, but I can't select any context now:

![2025-08-06 09 42 15](https://github.com/user-attachments/assets/3dba0c13-e572-4dd2-a3f6-7a6d550af1e8)

It would be great if we added a story for this regression (or even e2e).

Commit:
Merge remote-tracking branch 'origin/master' into fix/remove-auto-add-ui-context

Commit:
fix: do not reset context when setting chat url param

Commit:
fix: rename story

kappa90 commented:
> The context doesn't automatically attach, but I can't select any context now:
> 
> ![2025-08-06 09 42 15](https://private-user-images.githubusercontent.com/13541795/474887933-3dba0c13-e572-4dd2-a3f6-7a6d550af1e8.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ0NjY1NTcsIm5iZiI6MTc1NDQ2NjI1NywicGF0aCI6Ii8xMzU0MTc5NS80NzQ4ODc5MzMtM2RiYTBjMTMtZTU3Mi00ZGQyLWEzZjYtN2E2ZDU1MGFmMWU4LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA2VDA3NDQxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyOWY0MjNkNWU4YjBkNmQzMDY1MGEwYzIyZGE1NzUzYzYyMDJmMGE4NzdiNTAwMGY1OGM3ZTIwNmRmYjAxYWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SNaWVLAgal0Nuyi4lvrBXQkjZ7ghwkwPrlg2pelkK8E) [ ![2025-08-06 09 42 15](https://private-user-images.githubusercontent.com/13541795/474887933-3dba0c13-e572-4dd2-a3f6-7a6d550af1e8.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ0NjY1NTcsIm5iZiI6MTc1NDQ2NjI1NywicGF0aCI6Ii8xMzU0MTc5NS80NzQ4ODc5MzMtM2RiYTBjMTMtZTU3Mi00ZGQyLWEzZjYtN2E2ZDU1MGFmMWU4LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA2VDA3NDQxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyOWY0MjNkNWU4YjBkNmQzMDY1MGEwYzIyZGE1NzUzYzYyMDJmMGE4NzdiNTAwMGY1OGM3ZTIwNmRmYjAxYWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SNaWVLAgal0Nuyi4lvrBXQkjZ7ghwkwPrlg2pelkK8E) ](https://private-user-images.githubusercontent.com/13541795/474887933-3dba0c13-e572-4dd2-a3f6-7a6d550af1e8.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ0NjY1NTcsIm5iZiI6MTc1NDQ2NjI1NywicGF0aCI6Ii8xMzU0MTc5NS80NzQ4ODc5MzMtM2RiYTBjMTMtZTU3Mi00ZGQyLWEzZjYtN2E2ZDU1MGFmMWU4LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA2VDA3NDQxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyOWY0MjNkNWU4YjBkNmQzMDY1MGEwYzIyZGE1NzUzYzYyMDJmMGE4NzdiNTAwMGY1OGM3ZTIwNmRmYjAxYWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SNaWVLAgal0Nuyi4lvrBXQkjZ7ghwkwPrlg2pelkK8E) [ ](https://private-user-images.githubusercontent.com/13541795/474887933-3dba0c13-e572-4dd2-a3f6-7a6d550af1e8.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ0NjY1NTcsIm5iZiI6MTc1NDQ2NjI1NywicGF0aCI6Ii8xMzU0MTc5NS80NzQ4ODc5MzMtM2RiYTBjMTMtZTU3Mi00ZGQyLWEzZjYtN2E2ZDU1MGFmMWU4LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA2VDA3NDQxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyOWY0MjNkNWU4YjBkNmQzMDY1MGEwYzIyZGE1NzUzYzYyMDJmMGE4NzdiNTAwMGY1OGM3ZTIwNmRmYjAxYWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SNaWVLAgal0Nuyi4lvrBXQkjZ7ghwkwPrlg2pelkK8E)
> 
> It would be great if we added a story for this regression (or even e2e).

Fixed, I found out the bug, which was also why the context was always added even after removing it, in prod.

When we start a new chat, the `chat` param is added to the URL, this would reset the context. So in prod, resetting the context would have meant re-adding the default context and removing any user-selected context, and in this PR, it would have meant removing all the context. It would then work from the second message onwards as the URL params don't change.

I added a story but the chat responses have to be mocked so it doesn't really cover for this edge case.

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

skoob13 commented:
I see in devtools that the UI context is attached, but it's displayed in the message:
![2025-08-07 14 17 28](https://github.com/user-attachments/assets/d10f0539-55e7-44fd-9ecd-c99af16be4bb)

Commit:
Merge remote-tracking branch 'origin/master' into fix/remove-auto-add-ui-context

kappa90 commented:
@skoob13 I can't replicate it?

<img width="519" height="311" alt="image" src="https://github.com/user-attachments/assets/0bbdff69-9052-46bc-9170-0bdf4ec23337" />

skoob13 reviewed (approved):
Works now!
